### PR TITLE
Generic Convolution: Fix mismatches due to non-aligned widths for FP1…

### PR DESCRIPTION
**Pull request by Prithviraj-R**
…6 & moved input_channel to constant buffer

### CODE REVIEW

File: tools/cross_runner/kernels/conv_nchw_fp16.cpp [hunk 1]
```diff
@@ -209,7 +210,7 @@ _GENX_ inline void store_output_wc8_as_nchw(SurfaceIndex surface [[type("buffer_
 	const uint32_t pixel_offset = byte_offset / OUTPUT_ELEMENT_SIZE;
 	const bool b_oc_unalign_byte_offset = ((output_channels - (pixel_offset % output_channels)) < BLOCK_OC);
 	const bool b_ow_unalign_byte_offset = ((BLOCK_W + (pixel_offset % output_width)) > output_width);
-
+	//printf("%d %d %d %d %f\n", output_width, byte_offset, w_chunk_id, output_channels, grf_chunk[0]);
```
Comment (kgrundy20) [inline tools/cross_runner/kernels/conv_nchw_fp16.cpp:213]: Do you intend to leave this debug printf?

File: tools/cross_runner/kernels/conv_nchw_fp16.cpp [hunk 2]
```diff
@@ -393,13 +384,13 @@ extern "C" _GENX_MAIN_ void convolution_nchw_nondpas(
 	const uint32_t thread_per_full_oc = (output_channels + BLOCK_OC - 1)/ BLOCK_OC;
 	const uint32_t batch_id = thg_2 / thread_per_full_oc;
 	const uint32_t oc_chunk_id = thg_2 % thread_per_full_oc;
-    const uint32_t input_batch_offset = batch_id * BLOCK_BATCH * input_width * input_height * INPUT_CHANNELS * sizeof(DT_IN);
+    const uint32_t input_batch_offset = batch_id * BLOCK_BATCH * input_width * input_height * input_channels * sizeof(DT_IN);
     const uint32_t input_w_chunk_offset = w_chunk_id * BLOCK_W * STRIDE_W;
     const uint32_t input_h_chunk_offset = h_chunk_id * BLOCK_H * stride_h;
-	
+	//printf("P : %d %d %d\n", thg_0, thg_1, thg_2);
```
Comment (kgrundy20) [inline tools/cross_runner/kernels/conv_nchw_fp16.cpp:390]: Another printf

File: tools/cross_runner/kernels/conv_nchw_fp16.cpp [hunk 3]
```diff
@@ -449,6 +440,7 @@ extern "C" _GENX_MAIN_ void convolution_nchw_nondpas(
 			accu_row_0.select<1, 1, 1, 1>(b, bw) = activation_function(activation_type, activation_input, (float)activation_alpha, (float)activation_beta);
         }
     }
+	//printf("%f ", accu_row_0[0][0]);
```
Comment (kgrundy20) [inline tools/cross_runner/kernels/conv_nchw_fp16.cpp:443]: And this one

Comment (Prithviraj-R) [reply tools/cross_runner/kernels/conv_nchw_fp16.cpp]: removed

File: tools/cross_runner/kernels/conv_nchw_fp16.cpp (hunk 2 above)
Comment (Prithviraj-R) [reply tools/cross_runner/kernels/conv_nchw_fp16.cpp]: removed

**Review by kgrundy20 — APPROVED** [review]

**Pull request merged**